pkg/virt-handler/heartbeat/heartbeat.go

```diff
@@ -71,6 +71,33 @@ func (h *HeartBeat) heartBeat(heartBeatInterval time.Duration, stopCh chan struc
 	// 1 minute with a 1.2 jitter + the time it takes for the heartbeat function to run (sliding == true).
 	// So the amount of time between heartbeats randomly varies between 1min and 2min12sec + the heartbeat function execution time.
 	wait.JitterUntil(h.do, heartBeatInterval, 1.2, true, stopCh)
+
+	//ensure that the node is getting marked as unschedulable when removed
+	h.labelNodeUnschedulable()
+}
+
+func (h *HeartBeat) labelNodeUnschedulable() (done chan struct{}) {
+	done = make(chan struct{})
+	go func() {
+		now, err := json.Marshal(metav1.Now())
+		if err != nil {
+			log.DefaultLogger().Reason(err).Errorf("Can't determine date")
+			return
+		}
+		var data []byte
+		data = []byte(fmt.Sprintf(`{"metadata": { "labels": {"%s": "%s", "%s": "%t"}, "annotations": {"%s": %s}}}`,
+			v1.NodeSchedulable, "false",
+			v1.CPUManager, false,
```

Comment:
I wonder if we should set the cpu manager label to false.
virt-handler merely reflects whether the CPU manager is enabled by kubelet. This might be confusing to see it as disabled while it's not.
On the other hand, it's not relevant without virt-handler. Perhaps the best is just to delete it if it's causes issues.

Reply:
Wait. I wonder if this makes sense. CPU Manager applies to pods outside of the KubeVirt context. @jean-edouard what do you think?

Reply:
We should probably consider adding a `kubevirt.io` prefix to the `cpumanager` label. @vladikr looks like our short-term temporary fix until k8s starts adding this may stay for quite some more time. :D

Reply:
Yes, right now kubevirt just "takes ownership" of the cpumanager label, which is pretty bad. We definitely need to add a kubevirt-specific prefix.
That being said, it's outside of the scope of this PR, and we already do things like set cpumanager to false on all nodes if the feature gate is not enabled, so this is probably fine...

Reply:
> That being said, it's outside of the scope of this PR, and we already do things like set cpumanager to false on all nodes if the feature gate is not enabled, so this is probably fine...

Yes agree!

Reply:
> We should probably consider adding a kubevirt.io prefix to the cpumanager label. @vladikr looks like our short-term temporary fix until k8s starts adding this may stay for quite some more time. :D

@rmohr yeah, sure, we should. 

> Yes, right now kubevirt just "takes ownership" of the cpumanager label, which is pretty bad. We definitely need to add a kubevirt-specific prefix.

@jean-edouard the CPU manager has been released for at least 10 k8s releases. It doesn't seem that they really care about adding this detection functionally, on the other hand, we are very much dependent on it.